app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java

```diff
@@ -240,15 +253,22 @@ public void onProgressUpdated(String progressText, int progress) {
                                     holder.progressBarDownload.setProgress(progress);
                                 }
                             };
-                    new DownloadApplicationAsyncTask(
+                    DownloadApplicationAsyncTask downloadApplicationAsyncTask = new
+                            DownloadApplicationAsyncTask(
                             context.getApplicationContext(),
                             holder.progressBarDownload,
                             holder.textDownloadProgress,
                             holder.btnInstall,
                             holder.btnDownload,
                             downloadCompleteCallback,
                             progressUpdateCallback
-                    ).execute(applicationVersion);
+                    );
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
```

Comment:
Will the user experience a different behaviour if the Android version is below HONEYCOMB?

Reply:
Multiple AsyncTask was not supported to run in parallel below HONEYCOMB.
If we still want to support those APIs which are below HONEYCOMB, we should not use AsyncTask for network call because they will execute serially. Instead we can use other mechanism, for instance, thread pool executor

Reply:
Users below HONEYCOMB will face exactly the same issue that we have without this fix.